pkg/api/norman/customization/aks/listers.go

```diff
@@ -326,3 +337,35 @@ func encodeOutput(result interface{}) ([]byte, int, error) {
 
 	return data, http.StatusOK, err
 }
+
+func KubernetesVersionsFromOrchestratorVersionProfiles(profiles ...containerservice.OrchestratorVersionProfile) ([]string, error) {
```

Comment:
I think these newly introduced funcs should be unexported, especially considering the fact they are called from unexported func. Also, generally, we want exported funcs to be followed by unexported ones (https://github.com/uber-go/guide/blob/master/style.md#function-grouping-and-ordering).

Bottom line, if we make these unexported we'll be good.

P.S. Totally unrelated but I wish there was a (reasonable) way to eliminate copy-paste of these 2 funcs (as minimal as it is).

Reply:
I agree with unexporting them.  I found that in order to make this one function the type assertion logic made it difficult to read and it just made more sense to duplicate them.

Reply:
Yup, totally agree. My comment was more about the inability to define an interface for fields in Go, only for funcs. This would have helped in this case.

Reply:
Generics are coming